### PR TITLE
WIP win_file: refactor to add link support and other features

### LINKED ISSUE #42466: win_file: fails if the folder already exists with the attributes hidden.
##### SUMMARY
If the folder on the Windows system the attributes hidden, the the role fails. The error message is that the folder does not exist.

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
win_file

##### ANSIBLE VERSION
```
ansible 2.6.1
```

##### CONFIGURATION
n/a

##### OS / ENVIRONMENT
Ubuntu/CentOS

##### STEPS TO REPRODUCE
Create a folder on a Windows device.
```ps1
 (get-item (new-item -Path  C:\test -ItemType directory)).attributes="Hidden"
```
Ansible Task
```yaml
- win_file:
    path: "C:\\test "
    state: directory
```

##### EXPECTED RESULTS
n/a

##### ACTUAL RESULTS
n/a

### LINKED ISSUE #43060: win_file support managing links and junction points
##### SUMMARY
Need to add link support in win_file to bring it it closer to parity with what file can do.

##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
win_file

##### ANSIBLE VERSION
```
2.7
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
Windows

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
This is a refactor of the win_file module to bring some of the features in the `file` module across for Windows. This PR does the following.

* Add support for managing symbolic links, junction points, hard links
* Add support for managing paths > 260 chars when using the `\\?\` prefix
* Add support for setting the owner/group on a file
* Add support for setting attributes on a file
* Add support for setting the timestamps on a file
* Add the ability to bring idempotency to touch when setting `access_time: preserve` and `write_time: preserve`
* Add a deprecation warning for `state: file` when the file does not exist. This will change in a future version and will create the file instead of just warning the user
* Add diff support
* Cover more scenarios in the integration tests

Supersedes https://github.com/ansible/ansible/pull/28365
Fixes https://github.com/ansible/ansible/issues/42466
Fixes https://github.com/ansible/ansible/issues/43060

This PR relies on https://github.com/ansible/ansible/pull/43666 to be merged. The tests will fail until this is done.

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request

##### COMPONENT NAME
win_file

##### ANSIBLE VERSION
```
devel
```

### CODE REVIEW

**Comment by crossan007** [comment]
I am very much looking forward to this:
> Add support for managing symbolic links, junction points, hard links

File: lib/ansible/modules/windows/win_file.py [hunk 1]
```diff
@@ -12,43 +12,203 @@
...
+  access_time:
+    description:
+    - Set's the last access time attribute on the file.
+    - If not set, will default to C(now) when C(state=touch) and C(preserve)
+      for other states.
+    - If C(now), will set to the current datetime and is not idempotent.
+    - If C(preserve), will keep the existing value.
+    - Otherwise this is set to the datetime in the format specified by
+      I(access_time_format).
+    version_added: '2.7'
```
Comment (gundalow) [inline lib/ansible/modules/windows/win_file.py:39]: Doesn't look like this got merged intime, so `2.8` here (and other places

**Review by gundalow — COMMENTED** [review]
Will need moving to porting_guide_2.8
and all version + deprecation versions will need `n+1` version

**Comment by samccann** [comment]
@jborean93 - _gentle nudge_ - is this still an active PR in your queue?

**Comment by jborean93** [comment]
Somewhat but I’m taking a different approach but alas I have too many things going on at the same time :(

**Pull request closed without merging**